core/dbt/include/global_project/macros/adapters/timestamps.sql

```diff
@@ -0,0 +1,44 @@
+{%- macro current_timestamp() -%}
+    {{ adapter.dispatch('current_timestamp', 'dbt')() }}
+{%- endmacro -%}
+
+{% macro default__current_timestamp() %}
+    current_timestamp
+{% endmacro %}
+
+{%- macro current_timestamp_in_utc() -%}
+    {{ adapter.dispatch('current_timestamp_in_utc', 'dbt')() }}
+{%- endmacro -%}
+
+{% macro default__current_timestamp_in_utc() %}
+    {{ default__current_timestamp() }} at time zone 'utc'
+{% endmacro %}
+
+{%- macro snapshot_get_time() -%}
+    {{ adapter.dispatch('snapshot_get_time', 'dbt')() }}
+{%- endmacro -%}
+
+{% macro default__snapshot_get_time() %}
+    {{ current_timestamp() }}
+{% endmacro %}
+
+{%- macro convert_timezone(timestamp,  target_tz, source_tz) -%}
+    {%- if not target_tz is string -%}
+        {{ exceptions.raise_compiler_error("'target_tz' must be a string") }}
+    {%- else -%}
+        {{ adapter.dispatch('convert_timezone', 'dbt') (column, target_tz, source_tz) }}
+    {%- endif -%}
+
+{%- endmacro -%}
+
+{%- macro default__convert_timezone(column, target_tz, source_tz) -%}
```

Comment:
This takes Postgres's syntax as the default, which is often a fair play. In this case, the majority of other databases support `convert_timezone(source_tz, target_tz, column)` — so it probably makes more sense to have that one by our default in `dbt-core`, and reimplement just for `postgres__`.

_yes, `source_tz` is optional, doesn't it just delight you that the _optional_ argument is the first positional one??_

Reply:
gotcha, will update, let's definitely avoid for the future